designsafe/static/scripts/ng-designsafe/services/project-service.js

```diff
@@ -120,88 +120,188 @@ export function ProjectService(httpi, $interpolate, $q, $state, $uibModal, Loggi
+            input: 'Input',
+            output: 'Output',
+            hybrid_simulation: 'Hybrid Simulation',
+            global_model: 'Global Model',
+            coordinator: 'Coordinator',
+            sim_substructure: 'Simulation Substructure',
+            exp_substructure: 'Experimental Substructure',
+            mission: 'Mission',
+            collection: 'Collection',
+            planning: 'Research Planning Collection',
+            social_science: 'Social Sciences Collection',
+            geoscience: 'Engineering/Geosciences Collection',
+            analysis: 'Analysis',
+            report: 'Report',
+        };
 
+        let checkEntityHasFiles = (e) => {
+            if (!selFileListings[e.uuid]) {
+                return false;
+            }
+            return true;
+        };
+        
+        let addMissing = (missingEnt, fields) => {
+            if (!fields){
+                missingData.push({
+                    'title': missingEnt.value.title,
+                    'missing': ['Associated files/data are missing or not selected'],
+                    'type': errMsg[missingEnt.name.split('.').pop()]
+                });
+            } else {
+                let readableFields = fields.map(f => errMsg[f]);
+                missingData.push({
+                    'title': missingEnt.value.title,
+                    'missing': readableFields,
+                    'type': errMsg[missingEnt.name.split('.').pop()]
+                });
+            }
+        };
+
+        let checkRequirements = (set, subsets, required) => {
+            if (!subsets && !required) {
+                set.forEach((s) => {
+                    if (!checkEntityHasFiles(s)) {
+                        addMissing(s);
+                    }
+                });
+            } else {
+                set.forEach((s) => {
+                    let related =  subsets.filter(subset => subset.associationIds.includes(s.uuid));
+                    let checklist = [];
+
+                    related.forEach((relEnt) => {
+                        if (!checkEntityHasFiles(relEnt)) {
+                            addMissing(relEnt);
+                        } else if (!checklist.includes(relEnt.name.split('.').pop())) {
+                            checklist.push(relEnt.name.split('.').pop());
+                        }
+                    });
+                    let missing = required.filter(req => !checklist.includes(req));
+                    if (missing.length) {
+                        addMissing(s, missing);
+                    }
+                });
+            }
         };
 
-        let requiredSets = [];
-        let missingData = [];
-        let requiresFiles = ['analysis_set', 'report_set']; // checks if files are provided if included....
         if (project.value.projectType === 'experimental') {
-            requiredSets = ['modelconfig_set', 'sensorlist_set', 'event_set'];
+            /* 
+            Experimental Requirements:
```

Comment:
These comments spelling out the requirements for each project are super helpful

Reply:
Good to hear! :D